js/csv.js

```diff
@@ -0,0 +1,227 @@
+/**
+ * Reads a CSV file and returns a table with the data.
+ *
+ * @param {Blob|File} file The file object to be read.
+ * @param {Function} parser The function to be used in the
+ * CSV parsing operation.
+ */
+export const parseCsvFile = (file, parser = null) => {
+    parser = parser || parseCsvComplex;
+    return new Promise((resolve, reject) => {
+        const reader = new FileReader();
+        reader.onload = e => resolve(parser(e.target.result));
+        reader.onerror = e => reject(e);
+        reader.readAsText(file);
+    });
+};
+
+/**
+ * Creates the contents of the CSV file with the data given
+ * as an array of rows and the headers.
+ *
+ * @param {Array / Object} data Data to be converted into CSV. Can be
+ * an array of strings, a JSON object, or an array of those.
+ * @param {Array} headers List of headers for the CSV file, is an array of strings.
+ * @param {String} delimiter The delimiter to be used in the CSV.
+ * @returns A string containing the information for the CSV.
+ */
+export const buildCsv = (data, headers = [], delimiter = ",") => {
+    let csv = "";
+    if (Object.keys(headers).length !== 0) csv += headers.join(delimiter) + "\n";
+
+    // is simple object
+    if (!Array.isArray(data)) {
+        return csv + _objectToCsv(data, headers, delimiter);
+    }
+
+    // is simple array
+    if (typeof data[0] !== "object") return csv + _arrayToCsv(data, delimiter);
+    if (Object.keys(data).length === 0) return "";
+
+    for (let i = 0; i < Object.keys(data).length; i++) {
+        // check if array, if it is add value as is
+        if (Array.isArray(data[i])) {
+            csv += _arrayToCsv(data[i], delimiter);
+            continue;
+        }
+
+        csv += _objectToCsv(data[i], headers, delimiter);
+    }
+
+    return csv;
+};
+
+/**
+ * Converts an object into a CSV.
+ *
+ * @param {Object} data The object to be converted.
+ * @param {Array} headers The headers that are going to be
+ * searched for. If not existent, use keys for current
+ * data being converted.
+ * @param {String} delimiter The delimiter for the CSV.
+ * @returns A string of the converted object.
+ */
+export const _objectToCsv = (data, headers, delimiter = ",") => {
+    const fields = Object.keys(headers).length === 0 ? Object.keys(data) : headers;
+
+    return (
+        fields
+            .map(field =>
+                data[field] === null || data[field] === undefined
+                    ? ""
+                    : _parseStringDelimiter(data[field], delimiter)
+            )
+            .join(delimiter) + "\n"
+    );
+};
+
+/**
+ * Converts an array into a CSV.
+ *
+ * @param {Object} data The array to be converted.
+ * @param {String} delimiter The delimiter for the CSV.
+ * @returns A string of the converted array.
+ */
+export const _arrayToCsv = (data, delimiter = ",") => {

```

Comment:
It was a way to remove duplication of logic, since it was used in both [here](https://github.com/ripe-tech/ripe-commons-js/blob/d9bdb499cd23e0a311e7bbf1c9c62cb4ec3cf8d2/js/csv.js#L38) and [here](https://github.com/ripe-tech/ripe-commons-js/blob/d9bdb499cd23e0a311e7bbf1c9c62cb4ec3cf8d2/js/csv.js#L44)

Reply:
Note for documentation, the discussion here will be the same for this one: https://github.com/ripe-tech/ripe-commons-js/pull/33#discussion_r853835521.